File: engine/model/src/main/scala/tools/packet/playerinteraction/TradeChat.scala
package tools.packet.playerinteraction

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class TradeChat(private var _name: String, private var _chat: String, private var _owner: Boolean) extends PacketInput {
  def name: String = _name

  def chat: String = _chat

  def owner: Boolean = _owner

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/TradeChat.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record TradeChat(String name, String chat, Boolean owner) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}